<a href="https://colab.research.google.com/github/ob3dd/Adult-Income-and-Wine-Quality-/blob/main/Adult_Income_%26_Wine_Quality_Data_set.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##**Import**

In [1]:
import pandas as pd

##**First Choice Dataset**

##**Load Data**

In [8]:
fpath = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week08/adult.csv'
df = pd.read_csv(fpath)
df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   age              48842 non-null  int64 
 1   workclass        48842 non-null  object
 2   fnlwgt           48842 non-null  int64 
 3   education        48842 non-null  object
 4   educational-num  48842 non-null  int64 
 5   marital-status   48842 non-null  object
 6   occupation       48842 non-null  object
 7   relationship     48842 non-null  object
 8   race             48842 non-null  object
 9   gender           48842 non-null  object
 10  capital-gain     48842 non-null  int64 
 11  capital-loss     48842 non-null  int64 
 12  hours-per-week   48842 non-null  int64 
 13  native-country   48842 non-null  object
 14  income           48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

### **1. Source of data**

My source of data is https://www.kaggle.com/datasets/wenruliu/adult-income-dataset?resource=download

###**2. Brief Description of Data**

This dataset is about the income made by adults from all kinds of demographic. Giving an individual's annual income from various factors, but also influenced by their paths across life.

###**3. The Target**

Our Income column is our target.

###**4. Representation of a Particular Row**

The **"occupation"** row represents what kind of Jobs these individuals do to make the income that they make annually.

###**5. Classification or Regression Problem?**

This is a regression problem.

###**6. How many features does the data have?**

There are 15 feautures

###**7. How many rows are in the dataset?**

There are 48842 rows in the dataset

##**8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?**

There'd be a few
- Unnecessary Data
- Inconsistencies
- Irregular Datas

## **Second Choice Dataset**

##**Load Data**

In [9]:
fpath = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week08/wines_SPA.csv'
df_wine = pd.read_csv(fpath)
df_wine.info()
df_wine.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7500 entries, 0 to 7499
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   winery       7500 non-null   object 
 1   wine         7500 non-null   object 
 2   year         7498 non-null   object 
 3   rating       7500 non-null   float64
 4   num_reviews  7500 non-null   int64  
 5   country      7500 non-null   object 
 6   region       7500 non-null   object 
 7   price        7500 non-null   float64
 8   type         6955 non-null   object 
 9   body         6331 non-null   float64
 10  acidity      6331 non-null   float64
dtypes: float64(4), int64(1), object(6)
memory usage: 644.7+ KB


winery           wine  year  rating  num_reviews country  \
0  Teso La Monja          Tinto  2013     4.9           58  Espana   
1         Artadi  Vina El Pison  2018     4.9           31  Espana   
2   Vega Sicilia          Unico  2009     4.8         1793  Espana   
3   Vega Sicilia          Unico  1999     4.8         1705  Espana   
4   Vega Sicilia          Unico  1996     4.8         1309  Espana   

             region   price                  type  body  acidity  
0              Toro  995.00              Toro Red   5.0      3.0  
1    Vino de Espana  313.50           Tempranillo   4.0      2.0  
2  Ribera del Duero  324.95  Ribera Del Duero Red   5.0      3.0  
3  Ribera del Duero  692.96  Ribera Del Duero Red   5.0      3.0  
4  Ribera del Duero  778.06  Ribera Del Duero Red   5.0      3.0

### **1. Source of data**

My source of data is https://www.kaggle.com/datasets/fedesoriano/spanish-wine-quality-dataset

###**2. Brief Description of Data**

This dataset is related to red variants of spanish wines. Giving the quality of each wine from all factors taken.

###**3. The Target**

Our "Wine" column is our target.

###**4. Representation of a Particular Row**

Our "winery" is the Winery name. Where each wine in the data set was established or made.

###**5. Classification or Regression Problem?**

This is a regression problem

###**6. How many features does the data have?**

Our data has 10 features

###**7. How many rows are in the dataset?**

It has 7500 rows.

##**8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?**

There'd be a few
- Unnecessary Data
- Inconsistencies
- Irregular Datas